server/src/test/java/org/elasticsearch/search/query/QueryStringIT.java

```diff
@@ -51,13 +53,28 @@
 
 public class QueryStringIT extends ESIntegTestCase {
 
+    private static int CLUSTER_MAX_CLAUSE_COUNT;
+
+    @BeforeClass
+    public static void createRandomClusterSetting() {
+        CLUSTER_MAX_CLAUSE_COUNT = randomIntBetween(500, 1500);
```

Comment:
Can we test with small numbers, this seems big for a simple ut.

Reply:
Sure, the test was running with the default of 1024 before, so I wanted to stay in that area but there's no need to. Will lower it to the hundreds if that sounds okay.